src/auxiliary/special_elixirs.jl

```diff
@@ -36,6 +36,16 @@ julia> redirect_stdout(devnull) do
 ```
 """
 function trixi_include(mod::Module, elixir::AbstractString; kwargs...)
+    # Check that all kwargs exist as assignments
+    code = read(elixir, String)
+    expr = Meta.parse("begin \n$code \nend")
```

Comment:
Does this work if the included code contains a `module`? I'm asking because of

```
julia> begin

       module Foo

       end

       end
ERROR: syntax: "module" expression not at top level
Stacktrace:
 [1] top-level scope
   @ REPL[1]:1
```

Reply:
I can do
```julia
trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_advection_amr_refine_twice.jl"))
```
even though this elixir contains a module.

Reply:
Interesting...

Reply:
Weird idea: Is it necessary to parse the full elixir or could we just search (with a regex or so accepting whitespace) for something of the form `key =`, where `key` are the keywords we are looking for? Or is this a bad idea?

Reply:
I guess there would be many corner cases that we have to be aware of. Like `^key=`, `\skey=`, `\(key=`, `,key=`, `;key=`. But I don't see anything else that could go wrong, other than forgetting a case.

OTOH, we've been using this version for quite a while now in TrixiParticles, and I haven't found a case where it didn't work.

Reply:
Okay 👍